compiler/luci/import/src/Importer.cpp

```diff
@@ -125,7 +126,14 @@ void convert_graph(const luci::GraphBuilderSource &source, luci::CircleReader &r
   // NOTE Origin is intentionally not provided for constants.
   for (uint32_t i = 0; i < tensors.size(); ++i)
   {
-    luci::CircleConst *const_node = luci::create_circleconst(&gb_context, i);
+    luci::CircleNode *const_node = nullptr;
+
+    // create copy of const data if data is 'String' or copying required by user
```

Comment:
Please share why `String` is treated separately.

Reply:
`create_circleconst` handles Strings separately because format of data is differ:

```cpp
template <>
void copy_data<loco::DataType::STRING>(const VectorWrapper<uint8_t> &raw_data,
                                       uint32_t num_elements, CircleConst *const_node)
{
  assert(const_node->sparsityparam() == nullptr);

  const auto *data = reinterpret_cast<const char *>(raw_data.data());
  const auto *i32d = reinterpret_cast<const int32_t *>(raw_data.data());

  // de-serialize string data
  //   int32_t count
  //   int32_t offsets[count + 1]
  //   string  values[count]
  assert(static_cast<uint32_t>(*i32d) == num_elements);
  i32d++; // skip count

  std::vector<int32_t> offsets;
  offsets.push_back(*i32d++);
  for (uint32_t i = 0; i < num_elements; ++i)
  {
    offsets.push_back(*i32d++);
  }
  assert(offsets.size() == num_elements + 1);

  const_node->size<loco::DataType::STRING>(num_elements);
  for (uint32_t i = 0; i < num_elements; ++i)
  {
    int32_t start = offsets[i];
    int32_t next = offsets[i + 1];

    std::string value(data + start, next - start);
    const_node->at<loco::DataType::STRING>(i) = value;
  }
}
```

So, I think better to make copy for Strings.